coalib/parsing/LineParser.py

```diff
@@ -82,6 +82,15 @@ def _parse(self, line):
             line,
             self.comment_separators)
         comment = unescape(comment)
+
```

Comment:
Suppose the statement is "C# is cool" or something like "#This is a comment", it won't match this regex. Basically you need an or instead of and. Using two separate regex [^ ]# and #[^ ] in if elif statements would do but if there is a better regex use it preferably.